prow/spyglass/artifacts_test.go

```diff
@@ -66,9 +66,7 @@ func TestSpyglass_ListArtifacts(t *testing.T) {
 			args: args{
 				src: "gs/test-bucket/logs/example-ci-run/404",
 			},
-			want: []string{
-				"build-log.txt",
-			},
+			want: []string{},
```

Comment:
I changed this testcase because `ListArtifacts()` gets the build log name from the job instead of assuming the name is "build-log.txt".  Because of this I thought the unit test couldn't actually test adding the missing build log name when it doesn't exist in persistent storage, but I just realized I can probably test this properly if I change the fake job agent to handle this.